<launch>

    <!-- Launch the rosserial node that interface ROS with Arduino -->
    <node pkg="rosserial_python" type="serial_node.py" name= "serial_interface" respawn="true" output="screen">
        <param name="port" value="/dev/ttyACM0" />
    </node>

    <!-- Launch the angles converter service -->
    <node pkg="arduinobot_controller" type="angles_converter.py" name="angles_converter" respawn="true" output="screen"/>

    <!-- Launch the controller_interface node that converts joint state messages in Arduino readable messages -->
    <node pkg="arduinobot_controller" type="basic_controller_interface.py" name="basic_controller_interface" respawn="true" output="screen"/>

    <!-- Launch the joint state publisher GUI to control the joint angles -->
    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui"/>

</launch>